src/aPSFs.jl

```diff
@@ -10,6 +10,43 @@ function apsf(::Type{MethodParaxial}, sz::NTuple, pp::PSFParams; sampling=nothin
     res
 end
 
+function apsf(::Type{MethodCZT}, sz::NTuple, pp::PSFParams; sampling=nothing, center_kz=false) 
+    scale = (1.7, 2.0, 1.0, 1.0) # A tuple of factors (one for each dimension) to zoom into during the czt.
+   #It is a four dimensional factor, the first and second dimension represent x and y pixel.
+   #c_xy = (c_x, c_y, fixed factor, fixed factor) fixed factor == 1 
+
+   if isnothing(sampling)
+       sampling = get_Ewald_sampling(sz, pp)
+       print("Sampling is $sampling \n")
+   end
+
+   sz, sampling = size_sampling_to3d(sz, sampling)
+   check_amp_sampling(sz, pp, sampling)
+   pupil = pupil_xyz(sz, pp, sampling) # field_xyz(big_sz,pp, sampling) .* aplanatic_factor(big_sz,pp,sampling) .* ft(jinc_r_2d(big_sz[1:2],pp, sampling=sampling) .* my_disc(big_sz[1:2],pp)) 
```

Comment:
Maybe I am missing a point here, but I thought the the puple has to be sampled different parameters, so that it fills the availabe number of pixels. This looks like the original propagation code to me?